### ISSUE TITLE
Port is hardcoded as 9090

### ISSUE THREAD

**Issue by ManoharShenoy** [ganskef/MoCuishle#13]
Proxy port should be made configurable using a property so that it can be used even when some other server is already running on port 9090

**Comment by ganskef** [comment]
Yes of course, a Java property might be the first idea, but I have to find a solution for the browser extensions too. So I've used it with this hard coded port all the years :-).

Again (see #11), `okproxy.InterceptionProxy` allows a given port as argument. It might be a better starting point for your needs.

**Comment by ManoharShenoy** [comment]
I made changes to JavaPlatform.getProxyPort() to accept "proxy.port" as a "System Property" so that I could invoke the proxy as java -Dproxy.port=<port value> -jar mocuishle-2.2.0-shaded.jar. However, if I pass any other port value except 9090, the code just hangs. It seems like it is trying to connect to port 9090 to verify something & cannot proceed. Do you have any pointers on what could be going wrong? Code changes are as below.

```
public class JavaPlatform implements IPlatform {
.....
	@Override
	public int getProxyPort() {
		String proxyPortStr = System.getProperty("proxy.port", "9090");
		int proxyPort = 9090;
		try {
			proxyPort = Integer.parseInt(proxyPortStr);
		} catch (NumberFormatException e) {
			log.info("Non-integer value passed for proxy.port property. Ignored");
		}
		log.info("Proxy will be available on port " + proxyPort);
		return proxyPort;
	}
....
}
```
FYI, I am using "MoCuishle Proxy" as a "Browser Proxy" thru the "Proxy Settings" (to inject JS files into a web page) and hence, browser extension is not a concern for me.
